### ISSUE TITLE
zorder property changes trace color

### ISSUE THREAD

**Issue by stephprobst** [plotly/plotly.js#7235]
The zorder property changes the color of stacked line traces. The following code pens demonstrate the problem. The only change between the various charts is, that the zorder property has been added to the traces. This should not have any impact on the colouring, but it does.

[Stacked line traces (working / without zorder)](https://codepen.io/Djeramon/pen/qBejEBo)

![image](https://github.com/user-attachments/assets/0dd59962-2c85-4bfa-9b1f-a5668ada25f1)

[Stacked line traces (broken / zorder=1 on first trace)](https://codepen.io/Djeramon/pen/XWvgJzZ)

![image](https://github.com/user-attachments/assets/cd3dbbf0-6d80-45a6-bd35-ad6eb8df269c)

[Stacked line traces (broken / zorder on all traces)](https://codepen.io/Djeramon/pen/zYgzxpj)

![image](https://github.com/user-attachments/assets/d3dae676-3bd5-4e1e-a300-4c2af176dfa4)

**Comment by stephprobst** [comment]
@archmoj: Another problem with the zorder property, tested against the CI output artefacts you provided us with. (https://output.circle-artifacts.com/output/job/07ec9652-70cc-4287-bfb0-c639ea6b2d39/artifacts/0/dist/plotly.min.js)

**Comment by archmoj** [comment]
The traces in the codepens have transparency. @alexcjohnson do you this may be the expected behavior or a bug?

**Comment by alexcjohnson** [comment]
Yes, this is clearly a bug. All of these traces have `y=[1,1,1]` so they should each get a height of 1 no matter what else is done.

It seems that setting `zorder` confuses them about which other traces they're stacked on - each trace gets the right total y values, so they definitely know at that point what's stacked below them, but then when it comes time to draw the fills they go to the wrong place.

**Comment by Braintelligence** [comment]
Not meaning to hijack here but maybe this helps fixing the bug for good:

When I set `zorder: 2` to a single trace then all of my `showline` on all of the axes are gone. They reappear only if I set `mirror: all` but then the opposite side also appears.
